aten/src/ATen/LegacyTHFunctionsCPU.cpp

```diff
@@ -332,6 +332,18 @@ Tensor _th_histc(const Tensor & self, int64_t bins, const Scalar& min, const Sca
 }
 
 std::tuple<Tensor &,Tensor &> _th_gels_out(const Tensor & self, const Tensor & A, Tensor & res1, Tensor & res2) {
+    TORCH_WARN_ONCE(
+      "torch.lstsq is deprecated in favor of torch.linalg.lstsq and will be removed in a future PyTorch release.\n",
+      "torch.linalg.lstsq has reversed arguments and does not return the QR decomposition in "
```

Comment:
How should people get the QR decomposition, if they want it?

Reply:
We have a function with a Cthulhu name, `torch.geqrf`, to get the same QR tensor that `torch.lstsq` (previously `torch.gels`) returns.

Reply:
Yep, we should just point people to it

torch.ryleh would be a good namespace

Reply:
I've edited this and pointed the users to `torch.linalg.qr`. This function is difficult enough as it is to point them to these crazy functions that we happen to have in the `torch.` namespace.

Reply:
OK; let's get started with this